docker/client.py

```diff
@@ -151,6 +151,18 @@ def _raise_for_status(self, response, explanation=None):
         except requests.exceptions.HTTPError as e:
             if e.response.status_code == 404:
                 raise errors.NotFound(e, response, explanation=explanation)
+            if e.response.status_code == 400:
+                if response.content.startswith("client is newer than server"):
+                    explanation = (
+                        'docker-py is using newer API version than docker'
```

Comment:
It seems a bit strange to me that a library would report itself in the error message.

Usually a library is consumed by some other application, which probably doesn't want to expose these details in error messages.

I don't really know how I feel about this change. I think a better solution might be to just include the error string in the docs, that way if someone hits the issue and they search for it, they'll find the docs about how to set the version.